### DISCUSSION TITLE
Defining translations for states other than using the static $name property?

### DISCUSSION THREAD

**Discussion by developman-akl** [spatie/laravel-model-states#166] [Q&A] [answered]
Hi,

Is there any way to provide translation for the states other than using the static `$name` property?
That way we save translated state names into the DB, and also the translations are returned by the `Spatie\ModelStates\Events\StateChanged` properties.

Could you please give some advice? Thanks in advance.

**Comment by brendt** [comment, marked as answer]
That's not possible. I'd recommend looking into the viewmodel pattern or some kind of state resource to do name mappings for frontend purposes.

**Reply by developman-akl** [reply]
@brendt I think it is still not ideal to store the `public static $name` as state name in the database instead of the class name. 
Storing the class names would allow to enter translations via the `$name`.

**Reply by brendt** [reply]
Using the class name is problematic when refactoring state classes. You'd also have to write DB migrations to update the relevant classnames and is a hassle.
